### ISSUE TITLE
Can't save from Row Editor.

### ISSUE THREAD

**Issue by sidney-pham** [psequel/psequel#181]
<img width="1223" alt="screen shot 2017-12-12 at 12 17 43 am" src="https://user-images.githubusercontent.com/9250546/33832897-e5782e98-ded1-11e7-932a-a25360e0779d.png">

- Background goes red like in #173 and (possibly as a result of this) pressing the 'Save' button does nothing. 
- Adding a new row works fine, but if you try to edit an existing row saving doesn't work.

**Comment by joeherm** [comment]
To add to this, I've tried downgrading to previous versions and I'm experiencing the same issue. Not sure if this is true of you @sidney-pham or the other posters, but I am running on machine that has High Sierra installed.

**Comment by sidney-pham** [comment]
I actually updated to High Sierra a few days before I noticed the issue, so that's quite possibly the cause.

**Comment by Pat-rice** [comment]
Same here, also on high sierra and cannot use this feature anymore. It worked fine before the macos upgrade.

**Comment by hyouuu** [comment]
Same here. Seems it's not maintained anymore - anyone has better suggestions on a db tool for mac?

**Comment by hyouuu** [comment]
Submitted an issue in http://www.psequel.com/report_bugs, hopefully it could get author's attention

**Comment by maverickscrew** [comment]
You can try @hyouuu tableplus.io it has free tier but pretty good.

**Comment by hyouuu** [comment]
Thanks @maverickscrew ! From first glance & use it's pretty good!

**Comment by hyouuu** [comment]
Seems you're the dev behind tableplus - good job!

**Comment by perttuim** [comment]
I'm having the same issue but with an error message `ERROR: source for a multiple-column UPDATE item must be a sub-SELECT or ROW() expression`.

**Comment by rIpteyed** [comment]
I'm also experiencing this issue on High Sierra v10.13.3

+1

**Comment by fdeage** [comment]
Isn't this caused by a new version of pg instead of High Sierra?

**Comment by noloman** [comment]
same here in High Sierra and PSequel 1.5.3

**Comment by adamreisnz** [comment]
Had similar issues, reporting `source for a multiple-column UPDATE item must be a sub-SELECT or ROW() expression` when trying to save a row from the UI. This is not on High Sierra.

Switched to TablePlus seems to work like a charm 👍

**Comment by et** [comment]
Any update here? I'd be happy to help test this.
Mac OS Sierra
PSequel 1.5.3

**Comment by carterfiber** [comment]
Same here. Using free version of tableplus and seems to work fine. 

Using macos High Sierra

**Comment by angherq** [comment]
+1 with high sierra

**Comment by et** [comment]
For anyone new commenting here, take @adamreisnz's advice:

# SWITCH TO [TABLE PLUS](https://tableplus.io/)

`brew cask install tableplus`

* It's free-ish.
* Supports postgres + many more databases
* Looks so fucking good on mac
* Has active development

Thank you to whoever developed psequel. It saved my pickle from some time ago, but it's time to pass the torch.

**Comment by zmcNotafraid** [comment]
I got this error 
```
ERROR: source for a multiple-column UPDATE item must be a sub-SELECT or ROW() expression
```
after I upgrade postgresql from 9.6.X to 10.X.So It may be caused by postgresql.Not mac system version

**Comment by jide** [comment]
The issue is due to the app trying to update value using parenthesis when only editing one column in Postgres 10+ https://www.postgresql.org/message-id/DzifLcHs-9wlp5kq6MPKTxu6zI8s5LhAEaVsrq_NRTuQTkw_EWrogrLgDhQ7i5W-cvaIPig0gH59FzBk7Lwr8FUvV7H0fLOYI8heueaoG1Y%3D%40protonmail.com.

If you edit 2 fields, you won't see the error.

It's so easy to fix, it's frustrating we can't submit a PR :)

**Comment by yoshrubin** [comment]
> The issue is due to the app trying to update value using parenthesis when only editing one column in Postgres 10+ https://www.postgresql.org/message-id/DzifLcHs-9wlp5kq6MPKTxu6zI8s5LhAEaVsrq_NRTuQTkw_EWrogrLgDhQ7i5W-cvaIPig0gH59FzBk7Lwr8FUvV7H0fLOYI8heueaoG1Y%3D%40protonmail.com.
> 
> If you edit 2 fields, you won't see the error.
> 
> It's so easy to fix, it's frustrating we can't submit a PR :)

This saved me, I just cut and pasted the value of another column into itself and it let me save

**Comment by tapz** [comment]
Is this ever going to be fixed?

**Comment by yoshrubin** [comment]
Ok after thinking about this issue I will try to explain why I think this actually makes sense and might not need to be fixed.
Technically anytime you make a change to a value you are supposed to change the timestamp the value was `updated_at` (assuming you have this value) and in that case, it actually makes sense that you should not be able to save without updating that value with `NOW()` as the new value of the `updated_at` column. 
Without that column, it is very hard to justify needing to edit more than one column at a time and I can't think of any justifications.

**Comment by tapz** [comment]
@yoshrubin I disagree. With a db editor, each field should be edited separately. Only apps should change the update_at column.

**Comment by tapz** [comment]
@yoshrubin Could you please fix this. This is driving me crazy in tables, which do not have an updated_at field. A really stupid choice to require changing two fields just because you yourself always do that. At least make it a configurable feature as most people really don't want it.

**Comment by yoshrubin** [comment]
> @yoshrubin Could you please fix this. This is driving me crazy in tables, which do not have an updated_at field. A really stupid choice to require changing two fields just because you yourself always do that. At least make it a configurable feature as most people really don't want it.

I mean anyone can make a PR to fix this, I personally stopped using the app for the time being but feel free to make a fix @tapz

**Comment by jide** [comment]
I do not see how one could submit a PR, the project is closed source.

**Comment by tapz** [comment]
@yoshrubin If you don't use your own app anymore, publish the source code and let others develop it further.

**Comment by yoshrubin** [comment]
@tapz I did not develop this nor have I ever written code for this project. 
I am not sure why anyone here thinks I have, and @jide thanks for pointing that put I did not even realize it was closed source. I use TablePlus it has a free tier and it works very well.

**Comment by tapz** [comment]
@yoshrubin Sorry, I thought from your first comment that this is a design decision. No idea who the real author is, but he/she has been very quiet...

**Comment by iaforek** [comment]
> If you edit 2 fields, you won't see the error.

OMG I can't believe this actually worked! Also I can't believe this hasn't been fixed up now!

**Comment by tapz** [comment]
@iaforek I already moved on to TablePlus. It's a much better db editor, which of course is because you have to pay $59.

**Comment by iaforek** [comment]
I can't believe this is still open and there is no fix yet. What a shame! Has everyone switched to TablePlus by now?

**Comment by tapz** [comment]
@iaforek The latest version was released in 2017. So, I don't think the app is maintained anymore. Time to move on.
